src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1875,12 +1875,21 @@ private bool isValidToken(CORINFO_MODULE_STRUCT_* module, uint metaTOK)
         private bool isValidStringRef(CORINFO_MODULE_STRUCT_* module, uint metaTOK)
         { throw new NotImplementedException("isValidStringRef"); }
 
-        private char* getStringLiteral(CORINFO_MODULE_STRUCT_* module, uint metaTOK, ref int length)
+        private int getStringLiteral(CORINFO_MODULE_STRUCT_* module, uint metaTOK, char* buffer, int size)
         {
+            Debug.Assert(size >= 0);
+
             MethodILScope methodIL = HandleToObject(module);
             string s = (string)methodIL.GetObject((int)metaTOK);
-            length = (int)s.Length;
-            return (char*)GetPin(s);
+
+            fixed (char* ptr = s)
```

Comment:
This can be `s.AsSpan(0, Math.Min(size, s.Length)).CopyTo(new Span<char>(buffer, size)))`